packages/strapi-plugin-content-type-builder/controllers/validation/types.js

```diff
@@ -86,6 +86,19 @@ const getTypeShape = (attribute, { modelType, attributes } = {}) => {
           .test(isValidUID),
         minLength: validators.minLength,
         maxLength: validators.maxLength.max(256).test(maxLengthIsGreaterThanOrEqualToMinLength),
+        options: yup.object().shape({
+          separator: yup.string(),
+          lowercase: yup.boolean(),
+          decamelize: yup.boolean(),
+          customReplacements: yup.array().of(
+            yup
+              .array()
+              .of(yup.string())
+              .min(2)
+              .max(2)
+          ),
+          preserveLeadingUnderscore: yup.boolean(),
+        }),
```

Comment:
> Thanks for adding the Doc ! waiting for @lauriejim to review the doc.
> 
> Have you made sure you can still edit a content type in the interface without loosing your options ? (files are written on edit, not sure the option will be preserved)

I've added the options to the validation schema, this ensures the options field doesn't get removed when the attribute is changed.